google/cloud/completion_queue_test.cc

```diff
@@ -112,14 +112,20 @@ TEST(CompletionQueueTest, ShutdownWithPending) {
   {
     CompletionQueue cq;
     std::thread runner([&cq] { cq.Run(); });
-    timer = cq.MakeRelativeTimer(ms(20)).then(
-        [](future<StatusOr<std::chrono::system_clock::time_point>> result) {
-          // Timer still runs to completion after `Shutdown`.
-          EXPECT_STATUS_OK(result.get().status());
-        });
-    EXPECT_EQ(std::future_status::timeout, timer.wait_for(ms(0)));
+    // In very rare conditions the timer would expire before we get a chance to
+    // call `cq.Shutdown()`, use a little loop to avoid this. In most runs this
+    // loop will exit after the first iteration. By my measurements the loop
+    // would need to run more than once every 10,000 runs. On the other hand,
```

Comment:
Should be expect, then, that we never reach `i == 5`?

Reply:
I think you are right, but I wrote a comment because it is weird to have an assertion line that.